src/test/java/com/application/bookingservice/repository/accommodation/AccommodationRepositoryTest.java

```diff
@@ -0,0 +1,48 @@
+package com.application.bookingservice.repository.accommodation;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+
+import com.application.bookingservice.model.Accommodation;
+import com.application.bookingservice.model.Address;
+import java.math.BigDecimal;
+import org.junit.jupiter.api.Test;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.test.autoconfigure.jdbc.AutoConfigureTestDatabase;
+import org.springframework.boot.test.autoconfigure.orm.jpa.DataJpaTest;
+import org.springframework.test.context.jdbc.Sql;
+
+@DataJpaTest
+@AutoConfigureTestDatabase(replace = AutoConfigureTestDatabase.Replace.NONE)
+class AccommodationRepositoryTest {
+    @Autowired
+    private AccommodationRepository accommodationRepository;
+
+    @Test
+    @Sql(scripts = "classpath:db/"
+            + "accommodations/add-accommodations-and-addresses-to-tables.sql",
+            executionPhase = Sql.ExecutionPhase.BEFORE_TEST_METHOD)
+    @Sql(scripts = "classpath:db/"
+            + "accommodations/delete-accommodations-and-addresses-from-tables.sql",
+            executionPhase = Sql.ExecutionPhase.AFTER_TEST_METHOD)
+    void findById_AccommodationWithAddress_ReturnAccommodation() {
+        Address address = new Address()
+                .setId(1L)
+                .setCountry("Ukraine")
+                .setCity("Lviv")
+                .setStreet("Bohdan Khmelnytsky")
+                .setBuilding(100);
```

Comment:
no need to fully initialize objects when you need just their ids
In accommodation you need only object address with id initialized. And you can assertEquals this expected accommodation with that you got from repository (actual) and it will equal